### PR TITLE
DOC : added some documentation to j1 referring to spherical_jn

### PR DESCRIPTION
**Pull request by jrmlhermitte**
I suggest adding some documentation referring to the spherical Bessel functions as to someone not too exposed to them may confuse them. Just added two reference lines, I think this is enough.

From  #6865 

thanks!

julien

### CODE REVIEW

**Comment by pv** [comment]
Note the comment at the beginning of `_ufuncs.pyx`. The text should be changed in `add_newdocs.py`, and then run `python generate_ufuncs.py`

**Comment by ordirules** [comment]
oops, just 'grep'ed and searched for line, didn't see that.
Changed thanks!

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -3568,10 +3568,12 @@ def add_newdoc(place, name, doc):
     two rational functions of degree 6/6 and 7/7.
 
     This function is a wrapper for the Cephes [1]_ routine `j0`.
+    Not to be confused with the spherical Bessel functions (see spherical_jn).
```
Comment (person142) [inline scipy/special/add_newdocs.py]: Nitpicks:

- Put backticks around `spherical_jn` to make it a link
- The addition should be a complete sentence. Something like "This function is a wrapper for the Cephes [1]_ routine `j0`. It should not be confused..."

**Comment by person142** [comment]
Don't forget to run `python generate_ufuncs.py` and commit the changes. Preferably in its own commit with message `GEN: special: run generate_ufuncs.py`.

**Comment by ordirules** [comment]
thanks @person142 , made the change.
Also added your instructions to the add_newdocs.py file, in case it may be helpful in the future.

**Comment by pv** [comment]
thanks, merged!

**Pull request merged**